### DISCUSSION TITLE
Read tables at initial stage (Time=0)

### DISCUSSION THREAD

**Discussion by mnabideltares** [KratosMultiphysics/Kratos#11037] [General]
Hi,

I have tables in the MDPA file and I need to read them. It is required to my implementation to access the initial values of that tables, namely at time=0 before I start the time steps. I used the following statement in the "Initialize" stage, but it returned only zero's

    const GeometryType& Geom = this->GetGeometry();
    rVariables.roughnessTemperature = Geom[0].FastGetSolutionStepValue(AIR_TEMPERATURE);

I tried to read them during the first time step in "InitializeSolutionStep"

    if (!mIsInitialised)  {
        const GeometryType& Geom = this->GetGeometry();
        rVariables.roughnessTemperature = Geom[0].FastGetSolutionStepValue(AIR_TEMPERATURE, 1);
        mIsInitialised = true;
    }

However, I still get zero's, and it seems the values are totally not read at time=0. 
Do you have any suggestion how to read the values from a table at time=0?

Thank you very much.

**Comment by loumalouomega** [comment]
Hello, unfortunately tables are not automatic (this is an eternal discussion), doing that implies a penalty in performance. So you need to check that the table is defined and then retrieve values from the table. Example:

~~~cpp
double temp = 275;
if (rModelPart.GetProperties(0).HasTable(TEMPERATURE, DENSITY)) {
	auto& r_table = rModelPart.GetProperties(0).GetTable(TEMPERATURE, DENSITY);
    return r_table.GetValue(temp)
} else {
    return rModelPart.GetProperties(0)[DENSITY];
}
~~~

Related with https://github.com/KratosMultiphysics/Kratos/pull/10916
